### PR TITLE
Added build cache and develocity plugin

### PR DESCRIPTION
**Pull request by Mr3zee**
**Subsystem**
Gradle

**Solution**
Added Develocity build cache and scans

### CODE REVIEW

File: gradle-conventions-settings/develocity/src/main/kotlin/conventions-develocity.settings.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+        capture {
+            fileFingerprints = true
+        }
+
+        buildScanPublished {
+            scanJournal.appendText("${Date()} — $buildScanUri — $startParameter\n")
+        }
+
+        publishing.onlyIf { true }
```
Comment (CristianGM) [inline gradle-conventions-settings/develocity/src/main/kotlin/conventions-develocity.settings.gradle.kts:34]: This is the only thing I'm worried. Every external contributor will try publish scans and fail with a reject.

Comment (Mr3zee) [reply gradle-conventions-settings/develocity/src/main/kotlin/conventions-develocity.settings.gradle.kts:34]: @CristianGM I will introduce a property to disable publication attempt

**Review by e5l — APPROVED** [review]
Could you also add a similar one to Ktor?

**Comment by Mr3zee** [comment]
@e5l check https://github.com/ktorio/ktor/pull/4311

**Pull request merged**